### ISSUE TITLE
assertionFailure in OutputRendering.swift:256

### ISSUE THREAD

**Issue by finestructure** [cpisciotta/xcbeautify#156]
I'm hitting a reproducible assertionFailure in `OutputRendering.beautify`. The parameters are

```
(lldb) p line
(String) "    /Applications/Xcode-15.0.0.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -Xlinker -reproducible -target arm64-apple-ios14.0 -r -isysroot /Applications/Xcode-15.0.0.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk -O0 -L$PWD/checkout/.derivedData/Build/Intermediates.noindex/EagerLinkingTBDs/Debug-iphoneos -L$PWD/checkout/.derivedData/Build/Products/Debug-iphoneos -L/Applications/Xcode-15.0.0.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/lib -F$PWD/checkout/.derivedData/Build/Intermediates.noindex/EagerLinkingTBDs/Debug-iphoneos -F$PWD/checkout/.derivedData/Build/Products/Debug-iphoneos -F/Applications/Xcode-15.0.0.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/Frameworks -iframework /Applications/Xcode-15.0.0.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/Frameworks -iframework /Applications/Xcode-15.0.0.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0."...
```

```
(lldb) p pattern
(XcbeautifyLib.Pattern) linkerDuplicateSymbolsLocation
```

and

```
(lldb) p line.captureGroup(with: pattern) as CaptureGroup
(XcbeautifyLib.LinkerDuplicateSymbolsCaptureGroup)  (reason = "/Applications/Xcode-15.0.0.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -Xlinker -reproducible -target arm64-apple-ios14.0 -r -isysroot /Applications/Xcode-15.0.0.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk -O0 -L$PWD/checkout/.derivedData/Build/Intermediates.noindex/EagerLinkingTBDs/Debug-iphoneos -L$PWD/checkout/.derivedData/Build/Products/Debug-iphoneos -L/Applications/Xcode-15.0.0.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/lib -F$PWD/checkout/.derivedData/Build/Intermediates.noindex/EagerLinkingTBDs/Debug-iphoneos -F$PWD/checkout/.derivedData/Build/Products/Debug-iphoneos -F/Applications/Xcode-15.0.0.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/Frameworks -iframework /Applications/Xcode-15.0.0.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/Frameworks -iframework /Applications/Xcode-15.0.0.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/"...)
```

I'll try to create a test with just a string with a `Parser` but perhaps the info above is enough to help debug this.

I can see the case

```swift
        case (.linkerDuplicateSymbolsLocation, let group as LinkerDuplicateSymbolsLocationCaptureGroup):
            return formatLinkerDuplicateSymbolsLocation(group: group)
```

but the group is actually `LinkerDuplicateSymbolsCaptureGroup` here. So perhaps it just needs

```swift
        case (.linkerDuplicateSymbolsLocation, let group as LinkerDuplicateSymbolsCaptureGroup):
            return formatLinkerDuplicateSymbolsError(group: group)
```

?

**Comment by cpisciotta** [comment]
Hey @finestructure! I think `line`'s value might be truncated. If you look at the end, `SDKs/iPhoneOS17.0."...`, it appears there's more to the `String` that isn't shown. When I use a Regex tool, I'm unable to capture `line` using the `linkerDuplicateSymbolsLocation` pattern. Can you check if there is more to the `line` value, and - if possible - include a little more context? It might be helpful to post the raw log output if you have it. Thanks!

**Comment by cpisciotta** [comment]
One other thing... Is this actually an error around linker and duplicate symbols? I don't see anything that would indicate that.

**Comment by finestructure** [comment]
Thanks for taking a look, @cpisciotta !

It doesn't look like the line is actually truncated. That's just how the debugger's `p` command renders the (very long) string:

```
(lldb) p line.count
(Int) 2476
(lldb) p String(line.suffix(50))
(String) "/Products/Debug-iphoneos/TestTreeSitterLanguages.o"
```

Also note, that the regex _does_ actually match. The problem is that there's no case for the combination of a `.linkerDuplicateSymbolsLocation` and a `group` that's a `LinkerDuplicateSymbolsCaptureGroup`.

Looking at the full string, it seems to me that the problem is an erroneous match of the output line (which doesn't look like an error, just a `clang++` invocation):

```
(lldb) p print(line)
    /Applications/Xcode-15.0.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -Xlinker -reproducible -target arm64-apple-ios14.0 -r -isysroot /Applications/Xcode-15.0.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk -O0 -L/var/folders/nk/tlpts6bs799038htr6rx1xnc0000gn/T/AEE0B52D-8DEB-4D09-85F1-ED713ECE721E/checkout/.derivedData/Build/Intermediates.noindex/EagerLinkingTBDs/Debug-iphoneos -L/var/folders/nk/tlpts6bs799038htr6rx1xnc0000gn/T/AEE0B52D-8DEB-4D09-85F1-ED713ECE721E/checkout/.derivedData/Build/Products/Debug-iphoneos -L/Applications/Xcode-15.0.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/lib -F/var/folders/nk/tlpts6bs799038htr6rx1xnc0000gn/T/AEE0B52D-8DEB-4D09-85F1-ED713ECE721E/checkout/.derivedData/Build/Intermediates.noindex/EagerLinkingTBDs/Debug-iphoneos -F/var/folders/nk/tlpts6bs799038htr6rx1xnc0000gn/T/AEE0B52D-8DEB-4D09-85F1-ED713ECE721E/checkout/.derivedData/Build/Products/Debug-iphoneos -F/Applications/Xcode-15.0.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/Frameworks -iframework /Applications/Xcode-15.0.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/Frameworks -iframework /Applications/Xcode-15.0.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk/Developer/Library/Frameworks -filelist /var/folders/nk/tlpts6bs799038htr6rx1xnc0000gn/T/AEE0B52D-8DEB-4D09-85F1-ED713ECE721E/checkout/.derivedData/Build/Intermediates.noindex/Runestone.build/Debug-iphoneos/TestTreeSitterLanguages.build/Objects-normal/arm64/TestTreeSitterLanguages.LinkFileList -nostdlib -Xlinker -object_path_lto -Xlinker /var/folders/nk/tlpts6bs799038htr6rx1xnc0000gn/T/AEE0B52D-8DEB-4D09-85F1-ED713ECE721E/checkout/.derivedData/Build/Intermediates.noindex/Runestone.build/Debug-iphoneos/TestTreeSitterLanguages.build/Objects-normal/arm64/TestTreeSitterLanguages_lto.o -Xlinker -export_dynamic -Xlinker -no_deduplicate -Xlinker -debug_variant -Xlinker -dependency_info -Xlinker /var/folders/nk/tlpts6bs799038htr6rx1xnc0000gn/T/AEE0B52D-8DEB-4D09-85F1-ED713ECE721E/checkout/.derivedData/Build/Intermediates.noindex/Runestone.build/Debug-iphoneos/TestTreeSitterLanguages.build/Objects-normal/arm64/TestTreeSitterLanguages_dependency_info.dat -o /var/folders/nk/tlpts6bs799038htr6rx1xnc0000gn/T/AEE0B52D-8DEB-4D09-85F1-ED713ECE721E/checkout/.derivedData/Build/Products/Debug-iphoneos/TestTreeSitterLanguages.o
```

So it looks like my proposed fix isn't actually correct (although it fixes the crash for me, which is my main objective right now).

**Comment by finestructure** [comment]
I can't be 100% sure because we run a number of commands through xcbeautify when we encounter this error but FWIW I _think_ the source is the following command when running

```
env DEVELOPER_DIR=/Applications/Xcode-15.0.1.app xcrun xcodebuild docbuild OTHER_DOCC_FLAGS=--emit-digest -scheme Runestone -destination generic/platform=ios
```

on the repo https://github.com/simonbs/Runestone at revision 0.3.2.

**Comment by cpisciotta** [comment]
@finestructure I've released [1.0.1](https://github.com/tuist/xcbeautify/releases/tag/1.0.1). That should prevent the assertionFailure until there's a more proper fix.

**Comment by finestructure** [comment]
Fantastic, thank you!
